File: luaui/data/DataSourceFactory.lua
--[[
    DataSourceFactory.
    Faithful port of haxe_ui.data.DataSourceFactory.
--]]

local DataSourceFactory = {}
DataSourceFactory.__index = DataSourceFactory

function DataSourceFactory.new()
    local self = setmetatable({}, DataSourceFactory)
    return self
end

function DataSourceFactory:create(typeClass)
    return typeClass.new()
end

function DataSourceFactory:fromString(data, typeClass)
    local ds = self:create(typeClass)
    -- Logic to convert XML/JSON string to objects and add to data source
    return ds
end

function DataSourceFactory:fromStringToArray(data)
    -- Logic to convert XML/JSON string to array of objects
    return {}
end

return DataSourceFactory

File: flutter_ui/lib/data/data_source_factory.dart
class DataSource<T> {
  void add(dynamic o) {}
}

class DataSourceFactory<T> {
  DataSourceFactory();
  DataSource<T> create(Type type) {
    return DataSource<T>();
  }

  DataSource<T> fromString(String data, Type type) {
    var ds = create(type);
    if (data.startsWith("<")) {
      // XML parsing logic to be refined when XML library is added
    } else if (data.startsWith("[")) {
      // JSON parsing logic
    }
    return ds;
  }

  List<T> fromStringToArray(String data) {
    return [];
  }
}
